api/src/main/java/org/bahmni/module/events/api/publisher/EventPublisher.java

```diff
@@ -25,10 +27,29 @@ public EventPublisher(JmsTemplate jmsTemplate, ObjectMapper objectMapper) {
 	@EventListener
 	public void onApplicationEvent(Event event) {
 		String jsonPayload = toJsonPayload(event.payload);
-		jmsTemplate.send(event.eventType.topic(), session -> session.createTextMessage(jsonPayload));
+		jmsTemplate.send(event.eventType.topic(), session -> {
+			TextMessage message = session.createTextMessage(jsonPayload);
+			addMetaInfoInHeaders(message, event);
+			return message;
+		});
 		log.info("Published Message with id : " + event.payloadId);
 	}
 
+	private void addMetaInfoInHeaders(TextMessage message, Event event) {
+		try {
+			message.setStringProperty("eventType", event.eventType.name());
```

Comment:
I think there is no unit testing checking if MetaInfo in Headers is correctly set.
Also think the properties like eventId, payloadID, etc should not be "strings", because everyone who posts events needs to ensure the same metadata attributes are added. Modifying these attributes is RISKY once events are published / consumed. Introducing a type called: EventMetadataAttributeEnum might be good.

Reply:
Agree. Will do these changes shortly

Reply:
Done the changes introduced enums fro headers. Also I have written test to add headers as in I have introduced new class as JMSMessageCreator to create the message.